dirty_cat/_fuzzy_join.py

```diff
@@ -187,89 +317,125 @@ def fuzzy_join(
             f"how should be either 'left' or 'right', got {how!r}",
         )
 
-    if not isinstance(match_score, (int, float)):
-        raise TypeError("match_score has invalid type, expected integer or float")
+    if numerical_match not in ["string", "number", "error"]:
+        raise ValueError(
+            "numerical_match should be either 'string', 'number', or 'error', got"
+            f" {numerical_match!r}",
+        )
 
     for param in [on, left_on, right_on]:
-        if param is not None and not isinstance(param, str):
+        if param is not None and not (isinstance(param, Iterable)):
             raise TypeError(
                 "Parameter 'left_on', 'right_on' or 'on' has invalid type, expected"
-                " string"
+                " string or list of column names"
             )
 
-    # TODO: enable joining on multiple keys as in pandas.merge
-    if on is not None:
-        left_col = on
-        right_col = on
-    elif left_on is not None and right_on is not None:
-        left_col = left_on
-        right_col = right_on
+    if not isinstance(match_score, (int, float)):
+        raise TypeError("match_score has invalid type, expected integer or float")
+
+    if isinstance(on, str) or (isinstance(left_on, str) and isinstance(right_on, str)):
+        if on is not None:
+            left_col = [on]
+            right_col = [on]
+        elif left_on is not None and right_on is not None:
+            left_col = [left_on]
+            right_col = [right_on]
+    elif isinstance(on, Iterable):
+        left_col = list(on)
+        right_col = list(on)
+    elif isinstance(left_on, Iterable) and isinstance(right_on, Iterable):
+        left_col = list(left_on)
+        right_col = list(right_on)
     else:
         raise KeyError(
             "Required parameter missing: either parameter"
-            "'on' or the pair 'left_on', 'right_on' should be specified."
+            " 'on' or the pair 'left_on', 'right_on' should be specified."
         )
 
     if how == "left":
         main_table = left.reset_index(drop=True).copy()
         aux_table = right.reset_index(drop=True).copy()
-        main_col = left_col
-        aux_col = right_col
+        main_cols = left_col
+        aux_cols = right_col
     else:
         main_table = right.reset_index(drop=True).copy()
         aux_table = left.reset_index(drop=True).copy()
-        main_col = right_col
-        aux_col = left_col
-
-    # Make sure that the column types are string and categorical:
-    main_col_clean = main_table[main_col].astype(str)
-    aux_col_clean = aux_table[aux_col].astype(str)
+        main_cols = right_col
+        aux_cols = left_col
 
     # Warn if presence of missing values
-    if main_table[main_col].isna().any():
+    if main_table[main_cols].isna().any().any():
```

Comment:
Double any?

Reply:
This is on purpose, as we are:
- first looking if there are any missing values column-wise,
- then we output True if any of the columns has a missing value.